src/components/common/Directors.vue

```diff
@@ -545,44 +587,55 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
   $refs!: Vue['$refs'] & {
     // form and components to appoint a new director:
     newDirectorForm: FormIF,
-    baseAddressNew: BaseAddressIF,
-    mailAddressNew: BaseAddressIF,
+    baseAddressNew: ComponentIF,
+    mailAddressNew: ComponentIF,
     // form and components to edit an existing director:
     // (there is only 1 at a time but it's still an array)
     editDirectorForm: Array<FormIF>,
-    baseAddressEdit: Array<BaseAddressIF>,
-    mailAddressEdit: Array<BaseAddressIF>
+    baseAddressEdit: Array<ComponentIF>,
+    mailAddressEdit: Array<ComponentIF>
   }
 
-  // Props passed into this component.
+  /** Effective date for querying current directors and appointing/ceasing them. */
   @Prop({ default: '' })
   private asOfDate: string
 
-  /**
-   * Indicates whether this component should be enabled or not.
-   */
+  /** Indicates whether this component should be enabled or not. */
   @Prop({ default: true })
   private componentEnabled: boolean
 
-  private directorEditInProgress: boolean = false;
+  // Local definitions of computed properties for static type checking.
```

Comment:
I simply moved these up from lower down in the file. (I like having class variables declared at the top.)